### PR TITLE
[ll] Extend pso functionality and work on gl backend [43/..]

### PR DESCRIPTION
**Pull request by msiglreith**
* core: Add further configuration possibilities for graphics pipelines (still not finished so far: multisampling, depth-stencil state and dynamic states)
* core: Split the setting of reference values (blend constants and stencil ref) into two commands
* glutin: Fix window backend after the merge of `Resources` and `Backend`
* gl: Implement color clearing based on the old implementation

### CODE REVIEW

File: src/backend/gl/src/lib.rs [hunk 1]
```diff
@@ -489,6 +473,9 @@ impl CommandQueue {
             self.state.vao = true
         }
 
+        // Reset indirect draw buffer
+        unsafe { gl.BindBuffer(gl::DRAW_INDIRECT_BUFFER, 0) };
```
Comment (kvark) [inline src/backend/gl/src/lib.rs]: that would surely confuse some older driver, we need to gate this by a device feature

File: src/core/src/pso/graphics.rs [hunk 2]
```diff
@@ -15,10 +15,19 @@
...
-use Primitive;
 use super::EntryPoint;
-use super::input_assembler::{AttributeDesc, VertexBufferDesc};
-use super::output_merger::{ColorTargetDesc, DepthStencilDesc};
+use super::input_assembler::{AttributeDesc, InputAssemblerDesc, VertexBufferDesc};
+use super::output_merger::{ColorInfo, DepthStencilDesc};
+
+// Vulkan:
+//  - SpecializationInfo not provided per shader
+//  - TODO: infer rasterization discard from shaders?
```
Comment (kvark) [inline src/core/src/pso/graphics.rs]: we absolutely could, yeah, but Vk portability would be a bit easier if we had this flag explicitly exposed

File: src/core/src/pso/graphics.rs [hunk 3]
```diff
@@ -37,35 +46,104 @@ pub struct GraphicsShaderSet {
...
-            shader_entries: shader_entries,
-            color_targets: Vec::new(),
-            vertex_buffers: Vec::new(),
-            attributes: Vec::new(),
-        }
-    }
+///
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature="serialize", derive(Serialize, Deserialize))]
+pub struct DepthBias {
```
Comment (kvark) [inline src/core/src/pso/graphics.rs]: I guess that makes us completely detach from `draw_state`?

Comment (msiglreith) [inline src/core/src/pso/graphics.rs]: Yes, more and more /:

File: src/core/src/pso/graphics.rs (hunk 2 above)
Comment (msiglreith) [inline src/core/src/pso/graphics.rs]: okay

File: src/backend/gl/src/lib.rs (hunk 1 above)
Comment (msiglreith) [inline src/backend/gl/src/lib.rs]: ah right, getting all them correct is hard! Thanks for reviewing!

**Pull request merged**